FormSubmissionAndValidation2/src/components/CheckBox.jsx

```diff
@@ -0,0 +1,33 @@
+import PropTypes from "prop-types";
+import { ErrorMessage } from "./ErrorMessage";
+import { userFormConfig } from "../helper/userHelper";
+import React from "react";
+
+export const CheckBox = React.memo(function CheckBox({fieldId, value, errorMessage, updateValue}) {
```

Comment:
React.memo is ensuring that Checkbox will only be re-rendered when any of its prop changes. If I don't use this the CheckBox would be rendered every time its parent is re-rendered.
not sure of exact case when this can be problematic. But one disadvantage is that it enforces comparison of prop from previous render and this can be an overhead.

Reply:
@SUMIT5321 - Please go through this - https://stackoverflow.com/questions/76340126/should-i-use-usememo-to-wrap-components-delivered-as-props-in-react-composition

I think there's some confusion about the way we use useMemo.